### PR TITLE
Add fee validation to send signed meta transaction

### PR DESCRIPTION
**Pull request by cducrest**
closes https://github.com/trustlines-protocol/relay/issues/351

### CODE REVIEW

File: relay/blockchain/delegate.py [hunk 1]
```diff
@@ -64,7 +65,9 @@ def calc_next_nonce(self, identity_address: str):
         except UnexpectedIdentityContractException:
             raise InvalidIdentityContractException
 
-    def calculate_fees_for_meta_transaction(self, meta_transaction: MetaTransaction):
+    def calculate_fees_for_meta_transaction(
+        self, meta_transaction: MetaTransaction
+    ) -> DelegationFees:
```
Comment (berndbohmeier) [inline relay/blockchain/delegate.py]: Something seems wrong here with the type hint. 
I wonder why this is not caught by mypy.

File: relay/blockchain/delegate.py [hunk 2]
```diff
@@ -64,7 +65,9 @@ def calc_next_nonce(self, identity_address: str):
...
         try:
             valid = self.delegate.validate_nonce(meta_transaction)
```
Comment (berndbohmeier) [inline relay/blockchain/delegate.py]: why do we validate the nonce here?

Comment (cducrest) [reply relay/blockchain/delegate.py]: That's the only thing we can validate and it should allow failing earlier if the nonce is wrong.

File: relay/blockchain/delegate.py (hunk 1 above)
Comment (cducrest) [reply relay/blockchain/delegate.py]: If I run `mypy --ignore-missing-imports relay tests` it does not raise any error. But my IDE complains on the line `return self.delegation_fees` by saying "Expected type 'DelegationFees', got 'List[DelegationFees]' instead "

Comment (cducrest) [reply relay/blockchain/delegate.py]: It also does not complain when I run `mypy --ignore-missing-imports relay/blockchain/delegate.py `

Comment (berndbohmeier) [reply relay/blockchain/delegate.py]: mmh, sad, we could maybe see if we can somehow upgrade or so, to see if that fixes it. 
I'll open an issue.

File: relay/blockchain/delegate.py (hunk 2 above)
Comment (berndbohmeier) [reply relay/blockchain/delegate.py]: yes, I think it is just a weird place to do that.

File: tests/chain_integration/test_delegate.py [hunk 3]
```diff
@@ -255,3 +258,13 @@ def test_delegated_transaction_invalid_identity_contract(
...
+
+
+def test_meta_transaction_fees_valid(delegate, meta_transaction):
+    """
+    Check that no exception is raised when validating a valid meta_transaction
+    """
+    delegation_fees = delegate.calculate_fees_for_meta_transaction(meta_transaction)
+    meta_transaction.fees = delegation_fees.value
+    meta_transaction.currency_network_of_fees = delegation_fees.currency_network
+    delegate.validate_meta_transaction_fees(meta_transaction)
```
Comment (berndbohmeier) [inline tests/chain_integration/test_delegate.py]: Where is the test here?

Comment (berndbohmeier) [reply tests/chain_integration/test_delegate.py]: nvm, it is said in the comment.

**Pull request merged**